unuran/src/methods/dau.c

```diff
@@ -685,7 +689,7 @@ _unur_dau_make_urntable( struct unur_gen *gen )
   ratio = GEN->urn_size / sum;
   for( i=0; i<n_pv; i++ ) {
     GEN->qx[i] = pv[i] * ratio;  /* probability rescaled        */
-    if (GEN->qx[i] >= 1.) {      /* rich strip                  */
+    if (GEN->qx[i] >= 1. - FLT_EPSILON) {      /* rich strip                  */
```

Comment:
Interestingly, the MRE in scipy/scipy#19359 has a floating-point error of magnitude `1e-14`. Slightly more than the double-precision floating-point epsilon. So, I used the single-precision floating-point epsilon to fix it. 32-bit systems would anyway require that so I think this should be fine here.

```python
In [2]: x = 0.0006523157208088715

In [3]: sum = 0

In [4]: for i in range(1533): sum += x

In [5]: sum
Out[5]: 1.0000000000000424

In [6]: sum - 1.0
Out[6]: 4.241051954068098e-14
```